### PR TITLE
Build packages by distro

### PR DESCRIPTION
**Pull request by gnawhleinad**
via `build_distro.sh <el6|el7|wheezy|jessie>`.

### CODE REVIEW

**Comment by zmarano** [comment]
Hi Daniel,
Can you fill in some details about the intention here? What issue are you trying to resolve? It looks like you just created a wrapper for the build scripts- but no additional functionality.

**Comment by gnawhleinad** [comment]
> Can you fill in some details about the intention here? What issue are you trying to resolve? It looks like you just created a wrapper for the build scripts- but no additional functionality.

The use-case is to build a specific distro (e.g. `jessie`) and not have to generate packages for the other distros (e.g. `el6`, `el7`, and `wheezy`).

**Comment by illfelder** [comment]
Hi Daniel,

If you would like to add this functionality, I think it makes sense to build it into the build_packages scripts. You can add an optional argument to the script that, if specified, will only build for the particular distro. Otherwise all distros will be built.

**Comment by gnawhleinad** [comment]
```bash
$ ./build_packages.sh -h
Usage: build_packages.sh [options]

Build the Python package for Linux daemons, scripts, and libraries.

OPTIONS:
  -h             Show this message
  -o DISTRO,...  Build only specified distros
```

--

```bash
$ ./build_packages.sh -o jessie
Debian packaging tools generally labels all files in /etc as config files, as mandated by policy, so fpm defaults to this behavior for deb packages. You can disable this default behavior with --deb-no-default-config-files flag {:level=>:warn}
Debian packaging tools generally labels all files in /etc as config files, as mandated by policy, so fpm defaults to this behavior for deb packages. You can disable this default behavior with --deb-no-default-config-files flag {:level=>:warn}
Created package {:path=>"google-compute-engine-jessie_2.2.2-0.1472837492_all.deb"}
```

--

```bash
$ ./build_packages.sh -o jessie,wheezy
Debian packaging tools generally labels all files in /etc as config files, as mandated by policy, so fpm defaults to this behavior for deb packages. You can disable this default behavior with --deb-no-default-config-files flag {:level=>:warn}
Debian packaging tools generally labels all files in /etc as config files, as mandated by policy, so fpm defaults to this behavior for deb packages. You can disable this default behavior with --deb-no-default-config-files flag {:level=>:warn}
Created package {:path=>"google-compute-engine-jessie_2.2.2-0.1472837504_all.deb"}
Debian packaging tools generally labels all files in /etc as config files, as mandated by policy, so fpm defaults to this behavior for deb packages. You can disable this default behavior with --deb-no-default-config-files flag {:level=>:warn}
Debian packaging tools generally labels all files in /etc as config files, as mandated by policy, so fpm defaults to this behavior for deb packages. You can disable this default behavior with --deb-no-default-config-files flag {:level=>:warn}
Created package {:path=>"google-compute-engine-wheezy_2.2.2-0.1472837504_all.deb"}
```

File: build_packages.sh [hunk 1]
```diff
@@ -13,9 +13,17 @@
...
+#/
+#/ Build the Python package for Linux daemons, scripts, and libraries.
+#/
+#/ OPTIONS:
+#/   -h             Show this message
+#/   -o DISTRO,...  Build only specified distros
 
-TIMESTAMP="$(date +%s)"
+function usage() {
+  grep "^#/" <"$0" | cut -c 4-
```
Comment (illfelder) [inline build_packages.sh]: grep '^#/' < "$0" | cut -c 4-

File: build_packages.sh [hunk 2]
```diff
@@ -48,10 +56,47 @@ function build_distro() {
     setup.py
 }
 
-# RHEL/CentOS
-build_distro 'el6' 'rpm' '/usr/lib/python2.6/site-packages'
-build_distro 'el7' 'rpm' '/usr/lib/python2.7/site-packages'
+TIMESTAMP="$(date +%s)"
+
+while getopts "ho:" OPTION; do
```
Comment (illfelder) [inline build_packages.sh]: Please use single quotes when you are not quoting a variable.

File: build_packages.sh [hunk 3]
```diff
@@ -13,9 +13,17 @@
...
+#/
+#/ Build the Python package for Linux daemons, scripts, and libraries.
+#/
+#/ OPTIONS:
+#/   -h             Show this message
+#/   -o DISTRO,...  Build only specified distros
 
-TIMESTAMP="$(date +%s)"
+function usage() {
+  grep '^#/' <"$0" | cut -c 4-
```
Comment (illfelder) [inline build_packages.sh]: Please add the space after the redirect.

File: build_packages.sh (hunk 2 above)
Comment (illfelder) [inline build_packages.sh]: s/"ho:"/'ho:'/

File: build_packages.sh [hunk 4]
```diff
@@ -48,10 +56,47 @@ function build_distro() {
...
+
+while getopts "ho:" OPTION; do
+  case "$OPTION" in
+    h)
+      usage
+      exit 2
+      ;;
+    o)
+      set -f
+      IFS=","
```
Comment (illfelder) [inline build_packages.sh]: s/","/','/

File: build_packages.sh [hunk 5]
```diff
@@ -48,10 +56,47 @@ function build_distro() {
...
+    build_distro 'el7' 'rpm' '/usr/lib/python2.7/site-packages'
+    ;;
+  wheezy) # Debian 7
+    build_distro 'wheezy' 'deb' '/usr/lib/python2.7/dist-packages'
+    ;;
+  jessie) # Debian 8
+    build_distro 'jessie' 'deb' '/usr/lib/python2.7/dist-packages'
+    ;;
+  *)
+    echo "Invalid build: ${build}"
```
Comment (illfelder) [inline build_packages.sh]: Might be nice to include the valid build options here - what do you think?

File: build_packages.sh [hunk 6]
```diff
@@ -48,10 +56,47 @@ function build_distro() {
...
+    build_distro 'el7' 'rpm' '/usr/lib/python2.7/site-packages'
+    ;;
+  wheezy) # Debian 7
+    build_distro 'wheezy' 'deb' '/usr/lib/python2.7/dist-packages'
+    ;;
+  jessie) # Debian 8
+    build_distro 'jessie' 'deb' '/usr/lib/python2.7/dist-packages'
+    ;;
+  *)
+    echo "Invalid build '${build}': Use 'el6', 'el7', 'wheezy', or 'jessie'."
```
Comment (illfelder) [inline build_packages.sh]: s/:/./

**Comment by illfelder** [comment]
Nice change - looks great! I'll merge after the CLA is signed.

File: build_packages.sh [hunk 7]
```diff
@@ -48,10 +56,47 @@ function build_distro() {
...
+if [ -z "$BUILD" ]; then
+  BUILD=('el6' 'el7' 'wheezy' 'jessie')
+fi
 
-# Debian
-build_distro 'wheezy' 'deb' '/usr/lib/python2.7/dist-packages'
-build_distro 'jessie' 'deb' '/usr/lib/python2.7/dist-packages'
+for build in "${BUILD[@]}"; do
+  case "$build" in
+  el6) # RHEL/CentOS 6
```
Comment (illfelder) [inline build_packages.sh]: Just noticed this - please indent alternatives by 2 spaces (like you did above).

**Comment by illfelder** [comment]
Any issues with the CLA?

**Comment by gnawhleinad** [comment]
@illfelder, sorry for the delay!

We (:octocat:) received

> Thanks, your CLA has been submitted.

Is there anything I can do to kick-off the `cla/google` build?

**Comment by illfelder** [comment]
I signed it!

**Comment by illfelder** [comment]
It looks like the CLA bot will recheck every comment. Since you signed a corporate CLA, it won't show as passing, but I can still merge.

**Comment by illfelder** [comment]
Confirmed GitHub has a corporate CLA. Thanks for the contribution!

Merging!

**Comment by illfelder** [comment]
I just noticed this is checking in to master instead of development. Mind submitting a PR against development instead? I'm going to undo these merges.

**Comment by illfelder** [comment]
GitHub feature request - disable pull requests against master! :)

**Pull request merged**